[WIP] Try funkyheatmap-js

mxposed opened the pull request.

rcannood commented:
Things we're thinking about improving:

- [ ] Allow sorting by column
- [ ] Allow filtering by selecting methods / metrics / datasets / baseline methods (e.g. https://github.com/observablehq/inputs )
- [ ] Improve logistics (importing the package using `require()`?)
- [ ] Add interactive hovers to the rounded rects

In the (near?) future:
- Allow showing citation key when hovering over method, dataset or metric name (can we piggy back on current implementation for [citations in quarto](https://quarto.org/docs/authoring/footnotes-and-citations.html))
- Allow displaying other kinds of meta data, such as normalization type, hvg selection.

mxposed commented:
I fixed the logistics (almost fully), and added tooltip on hover.
For tooltip a todo item would be to trigger tooltip based on area, rather than actual geom: for small circles it's hard to select them. Another todo item: highlight column and row labels.

rcannood commented:
I moved the remaining items to separate issues. If we change the tooltips I feel we can merge this @mxposed :relaxed:

rcannood commented on results/_generate_task_pages.py:
```diff
@@ -21,6 +21,10 @@
 engine: knitr
 image: featured.png
 page-layout: full
+format:
+    html:
+        header-includes: |
+              <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/d3/7.8.2/d3.js"></script>
```
```suggestion
```

mxposed commented:
I fixed the tooltips here https://github.com/mxposed/funkyheatmap-js/commit/c576163cd57be8fdf439da1dacad84012e6b2b6b
And also removed flickering when the `Rescale` toggle was clicked

scottgigante reviewed (changes requested):
* The rescale toggle needs to be either made less ugly or temporarily removed before merge.
* Removing the results table entirely means now there is nowhere on the website that shows the score of a given method on a given metric and dataset. This seems like a bad idea.

rcannood commented:
@mxposed IMO this is almost ready to be merged. One thing I noticed is that the svg renders outside the bounds of the div. 

![Uploading Screenshot from 2023-03-08 16-49-58.png…]()

scottgigante commented:
> You can see the scores by hovering over the FH:

That gives the mean score over all datasets. The only way I can see "what is the ARI for scANVI (hvg/unscaled) on the lung dataset" is by downloading the raw data and poring through a giant JSON file.

mxposed commented:
Well, the problem with overflow is this:
1. funkyheatmap-js renders its dimensions based on the content (so in this case because of long method names the content is wider than available space)
2. the cell rendering css does not have (currently) any overflow handling

I see 2 ways to deal with this:
1. add overflow-x: auto to cell css, then figure will stay the same, but will display cropped, available for scrolling
2. make funkyheatmap scale to certain width, or be constrained by it, and pass this as a parameter (it does not know its parent's width when rendering—ojs ways)
3. shorten method names (and/or add code to shorten method names automatically)

what do y'all think?

scottgigante commented:
> shorten method names (and/or add code to shorten method names automatically)

Already done on the v1 repo. I just ran an update to push those changes: https://github.com/openproblems-bio/website-experimental/pull/87

Re: overflow and scrolling, I think I'd prefer to make each section (name, overall, datasets, metrics) a maximum width and scroll within each rather than scrolling the entire figure? but open to disagreement. Short term overflow-x: auto seems fine.

mxposed commented:
overflow-x fix applied, see [here](https://6409081f065ae51749b0d5da--openproblems-experimental.netlify.app/results/batch_integration_feature/). The commit didn't trigger the build in this pull-request for some reason…

rcannood commented:
> The commit didn't trigger the build in this pull-request for some reason…

the github actions don't run when there are conflicts that need to be resolved ;)

rcannood commented:
Merging since Scott's concerns were addressed.
